File: internal/models/league.go
package models

import (
	"github.com/google/uuid"
	"time"
)

type League struct {
	ID   uuid.UUID `gorm:"type:uuid;default:gen_random_uuid();primary_key"`
	Name string    `gorm:"type:varchar(255);not null"`

	// TODO: Limit to 20 characters and update request validation accordingly
	Slug       string `gorm:"type:varchar(255);not null;uniqueIndex"`
	Owner      User
	OwnerID    uuid.UUID `gorm:"type:uuid;not null"`
	Location   Location
	LocationID uuid.UUID `gorm:"type:uuid;not null"`
	CreatedAt  time.Time
	UpdatedAt  time.Time
}

File: web/app/src/test/fake/league.ts
import {League} from "../../api";
import {faker} from "@faker-js/faker";
import * as helpers from "../../helpers";
import {location} from "./location";

export function league(): League {
  const name = faker.hacker.phrase();

  return {
    id: faker.datatype.uuid(),
    name: name,
    slug: helpers.slugify(name),
    location: location(),
    owner_id: faker.datatype.uuid(),
    created_at: faker.date.recent(5).toISOString(),
    updated_at: faker.date.recent(1).toISOString()
  }
}
